### DISCUSSION TITLE
correccion de datos inválidos

### DISCUSSION THREAD

**Discussion by ypne** [IIC2413/Syllabus-2024-2#86] [General]
Con respecto a la corrección de datos solicitada y esbozada en el flujo y psudocódigo, les entregamos una tercera guia complementaria.
1.- Bananer.php debe detectar todos los datos que violen las reglas del enunciado y escribir esas tuplas en un archivo separado al que llamaremos log.txt
2.- Luego se debe revisar las tuplas de log.txt y evaluar las acciones a seguir con el objetivo de poder cargar el máximo de información posible.
     IF la tupla se puede corregir THEN {
             IF la corrección se puede automatizar THEN se modifica bananer.php para que corrija todos los errores de este tipo
            ELSE se corrige manualmente 
     ELSE IF la tupla se puede marcar con un valor que permite procesarla sin afectar al resto (por ejemplo un teléfono inválido cambiarlo por 000000000) THEN se hace el cambio
      ELSE (si la tupla no se puede corregir ni marcar) se elimina la tupla y todas las relacionadas
3.- Se transfieren los cambios de las tuplas de log.txt no automatizables en los archivos de input. Con esto al correr por segunda vez Bananer.php no debería haber ninguna tupla inválida
4.- Se  cargan las tuplas en las matrices y realizan las consultas

**Comment by cataruze** [comment]
Es necesario hacerlo de esta forma? Con el archivo log.txt o si lo estoy haciendo de otra manera puedo continuar como estaba ?

**Reply by ypne** [reply]
puedes hacerlo de otra manera. Tanto este artículo como el flujo y el pseudocódigo son UNA posible solución a la tarea encomendada.
